src/main/java/uk/gov/hmcts/ccd/data/userprofile/UserProfileRepository.java

```diff
@@ -177,6 +177,38 @@ private boolean isAuditable(final UserProfileEntity entity) {
         return query.getResultList().stream().map(UserProfileMapper::entityToModel).collect(Collectors.toList());
     }
 
+    public List<UserProfileLight> findAllLight(String jurisdictionId, final String actionedBy) {
+
+        TypedQuery<UserProfileJurisdictionEntity> query1 = em.createNamedQuery(
+            "UserProfileJurisdictionEntity.findAllByJurisdiction", UserProfileJurisdictionEntity.class);
+        query1.setParameter("jurisdiction", jurisdictionId);
+
+        Set<String> profileIdsInJurisdiction = query1.getResultList().stream()
```

Comment:
Could this not be a single query rather than loading the IDs and then issuing a second query? It would also avoid having to create the new UserProfileJurisdictionEntity

Reply:
Tried to create a single query, but couldn't get it done with Hibernate quickly. Apparently there is another ticket that will override this and this is a temporary solution (should not say that, as it might stay for another year :) ), so we have agreed it is good enough.